### PR TITLE
Remove hard reset of radsan test arch

### LINKED ISSUE #21: `make generate` fails on macOS x86_64
I'm adding the `macos-latest` platform to our CI test workflow, and can reproduce [this failure](https://github.com/realtime-sanitizer/radsan/actions/runs/7268078880/job/19803235427) locally on my x86_64 iMac - it looks like our CMake configuration is attempting to create the radsan test targets twice!

### PR DESCRIPTION
**Pull request by davidtrevelyan**
It looks like we had a mistake in the loop over test architectures, where the CMake loop variable `arch` was overridden to a default. This [resulted](https://github.com/realtime-sanitizer/radsan/actions/runs/7268078880/job/19803235427) in trying to generate two test targets with exactly the same name, because this loop was running over `x86_64;x86_64h` architectures on intel macs!

### CODE REVIEW

File: compiler-rt/lib/radsan/tests/CMakeLists.txt
Comment (davidtrevelyan) [inline compiler-rt/lib/radsan/tests/CMakeLists.txt:57]: This is the key difference, the rest of the changes are just removing dead configuration code!

**Review by cjappl — APPROVED** [review]
good catch!

**Comment by cjappl** [comment]
please merge as you see fit. Thanks for the addition to this repo, I think we can now transition to the "original submitter submits once they have approval"

This removes at least one back and forth :)

**Comment by davidtrevelyan** [comment]
While this makes the Intel Mac configuration stage pass, unfortunately it hasn't fully fixed the Intel Mac build, which now fails with duplicate symbol errors. More investigations needed!

**Review by cjappl — APPROVED** [review]

**Pull request merged**
